storm-client/src/jvm/org/apache/storm/daemon/worker/WorkerState.java

```diff
@@ -186,6 +186,9 @@ public WorkerState(Map<String, Object> conf,
         this.isWorkerActive = new CountDownLatch(1);
         this.isTopologyActive = new AtomicBoolean(false);
         this.stormComponentToDebug = new AtomicReference<>();
+        this.topology = ConfigUtils.readSupervisorTopology(conf, topologyId, AdvancedFSOps.make(conf));
+        this.taskToComponent = StormCommon.stormTaskInfo(topology, topologyConf);
+        // mkReceiveQueueMap relies on taskToComponent which relies on topology above
```

Comment:
I would change `mkReceiveQueueMap` method to include `taskToComponent` parameter to make this relationship more obvious

Reply:
SUre